### PR TITLE
feat: new event to send emails to winners when the ranking finishes and fix in the organization of events

### PR DESCRIPTION
**Pull request by PatricioPoncini**
### `Changes`

- new event to send emails to winners when the ranking finishes
- fix in the organization of events

### `docs`

docs are in the respective files

### CODE REVIEW

**Comment by PatricioPoncini** [comment]
https://legendaryum.atlassian.net/jira/software/projects/LE/boards/1?assignee=712020%3A71750ee7-8648-475c-99c7-c39c60a8e3fc&selectedIssue=LE-2582

File: my-awesome-rabbitmq-lib/src/events.rs [hunk 1]
```diff
@@ -164,31 +194,31 @@ impl PayloadEvent for LegendMissionsOngoingMissionEventPayload {
...
 }
 
 #[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CoinsSendEmailPayload {
-    pub user_id: String,
-    pub email_type: String,
-    pub email: String,
-    pub coins: i32,
+pub struct LegendRankingsRankingsFinishedEventPayload {
```
Comment (jym272) [inline my-awesome-rabbitmq-lib/src/events.rs:212]: usa `#[serde(rename_all = "camelCase")]` al principio del struct

File: my-awesome-rabbitmq-lib/src/events.rs [hunk 2]
```diff
@@ -164,31 +194,31 @@ impl PayloadEvent for LegendMissionsOngoingMissionEventPayload {
...
+    pub title: String,
+    #[serde(rename = "rewardType")]
+    pub reward_type: RewardType,
+    pub winners: Vec<RankingWinner>,
 }
```
Comment (jym272) [inline my-awesome-rabbitmq-lib/src/events.rs:217]: Este no es el payload, el payload que declaraste en la librería de TS es `  completedRankings: CompletedRanking[];`, cambia la librería de TS

File: my-awesome-rabbitmq-lib/src/events.rs [hunk 3]
```diff
@@ -164,31 +194,31 @@ impl PayloadEvent for LegendMissionsOngoingMissionEventPayload {
...
+    Legends,
+    CodeExchange,
 }
 
-impl PayloadEvent for CoinsNotifyClientPayload {
-    fn event_type(&self) -> MicroserviceEvent {
-        MicroserviceEvent::CoinsNotifyClient
-    }
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RankingWinner {
```
Comment (jym272) [inline my-awesome-rabbitmq-lib/src/events.rs:205]: usa #[serde(rename_all = "camelCase")] al principio del struct

**Review by jym272 — CHANGES_REQUESTED** [review]

File: my-awesome-rabbitmq-lib/src/events.rs (hunk 2 above)
Comment (PatricioPoncini) [reply my-awesome-rabbitmq-lib/src/events.rs:217]: Rigth, here is the correction:
```.ts
// ts

// interface
export interface CompletedRanking {
    title: string;
    rewardType: RankingsRewardsType;
    winners: RankingWinners[];
}

// event
'legend_rankings.rankings_finished': {
      completedRankings: CompletedRanking[];
};
```

```rust
// .rs

// struct
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CompletedRanking {
    pub title: String,
    pub reward_type: RankingsRewardsType,
    pub winners: Vec<RankingWinners>,
}

// event
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LegendRankingsRankingsFinishedEventPayload {
    pub completed_rankings: Vec<CompletedRanking>,
}

**Pull request merged**